rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -2696,27 +2691,27 @@ boolean internalLockNode(RMNode rmNode, Client lockInitiator) {
      */
     @Override
     public BooleanWrapper unlockNodes(Set<String> urls) {
-        return mapOnNodeUrlSet(urls, new Predicate<RMNode>() {
-            @Override
-            public boolean apply(RMNode node) {
-                return internalUnlockNode(node);
-            }
-        }, "unlock");
+        return mapOnNodeUrlSet(urls, this::internalUnlockNode, "unlock");
     }
 
     public BooleanWrapper mapOnNodeUrlSet(Set<String> nodeUrls, Predicate<RMNode> operation, String operationName) {
         boolean result = true;
 
         for (String url : nodeUrls) {
-            RMNode rmnode = getNodeByUrlIncludingDeployingNodes(url);
+            try {
+                RMNode rmnode = getNodeByUrlIncludingDeployingNodes(url);
 
-            if (rmnode == null) {
-                logger.warn("Cannot " + operationName + ", unknown node: " + url);
-                result &= false;
-                continue;
-            }
+                if (rmnode == null) {
+                    logger.warn("Cannot " + operationName + ", unknown node: " + url);
+                    result = false;
+                    continue;
+                }
 
-            result &= operation.apply(rmnode);
+                result &= operation.apply(rmnode);
+            } catch (Exception e) {
+                logger.error("Exception during mapOnNodeUrlSet.", e);
```

Comment:
it would be better to log a similar error message as the above warning, with the exception